src/core/ComplexFilter/index.tsx

```diff
@@ -18,23 +18,29 @@ export type ComplexFilterValue =
   | DefaultMenuSelectOption
   | null;
 
-interface ComplexFilterProps {
+export type Value<T, Multiple> = Multiple extends undefined | false
```

Comment:
This is how we define a conditional type!

Reply:
Wow, this really feels like it's at the edge of my understanding. Let me describe it in my own words and see if that's correct:
- a `Value` type is taking two args: `T` and `Multiple`
- The `Multiple extends` is a truthy/falsy check on if the `Multiple` arg "matches" with `undefined | false`
- From there, we have a ternary that sets the resulting Value type based on that match -- truthy becomes `T | null`, while falsy becomes an array of `T` and still `null`.
- Above is because if Multiple "matches" with `undefined | false`, then it was not present in the incoming props, and thus we are in the single case, etc

All that seems to hold up pretty well. The big thing that really freaks me out here is the use of `extends` to do this kind of type checking? I thought `extends` was an imperative/declarative keyword _only_: you used it to inherit from "parent" types, ala OOO. [I can find docs and articles on this usage of `extends`](https://www.typescriptlang.org/docs/handbook/2/objects.html) pretty easily, but I can't find anything at all on this other meaning of `extends` as a "question-asking" form? In this usage it feels like a way of doing `typeof` inside of TS-only code. Did you find any docs on this usage of it?

Reply:
Also, maybe this isn't as complicated as it feels to me, but I think a line or two of comments above explaining what's going on here would be good. This seems to be a pretty unusual structure, and I think having a bit of plain English that explains why it exists and what it does would be helpful for those with less TS experience (cough cough me cough)

Reply:
Great idea! I added comments to explain the usage being [Conditional Type](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) 🚀 !